docs/azure/sdk/configure-proxy.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Configure a proxy when using the Azure SDK for .NET 
+description: Learn different approaches for configuring a proxy for use with the Azure SDK for .NET client libraries.
+ms.topic: conceptual
+ms.custom: devx-track-dotnet, engagement-fy23
+ms.date: 11/15/2024
+---
+
+# Configure a proxy when using the Azure SDK for .NET
+
+If your organization requires the use of a proxy server to access Internet resources, some configuration is required to use the Azure SDK for .NET client libraries. Once configured, the proxy is applied to the underlying `HttpClient` instance used for HTTP operations.
+
+The proxy can be configured via code or via an environment variable. The approach you choose depends on the desired behavior. Set the appropriate environment variable if you want the proxy to apply globally to all service clients created within the current process. Alternatively, configure the proxy via code to selectively apply the settings to service clients.
```

Comment:
I have not confirmed this empirically, but based on my read-through of the relevant Azure.Core code (see [HttpClientTransport.cs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs)), if the "configure via code" approach is used -- i.e. a custom transport is provided to a client via client options, and that transport is constructed from the HttpMessageHandler constructor overload on HttpClientTransport -- then the default settings applied to the HttpClient held by the default transport will not be applied.

I do not currently have a good sense of the implication of this, but I believe it is equivalent to setting a custom HttpClient on the transport, and overriding the default HttpClient.  The default HttpClient increases the number of connection limits for application performance reasons, overrides the HttpClient timeout so that the timeout specified on the client options will be used instead, turns off automatic redirects for security reasons, and a few other things related to cookies and auth certs that I believe are needed for a subset of Azure clients.

I don't have a specific recommendation at present regarding how to represent this in the docs, but it might be worth calling out that that using the "configuration via code" option will override Azure SDK HttpClient defaults, and that may cause interop problems with some Azure clients.  We can track down those specific details if needed.

Reply:
As discussed via Teams, we'll update the doc once we better understand the interop problems.